mlir/test/Dialect/Vector/vector-tofrom-elements-to-shuffle-tree-transforms.mlir

```diff
@@ -0,0 +1,329 @@
+// RUN: mlir-opt -lower-vector-to-from-elements-to-shuffle-tree -split-input-file %s | FileCheck %s
+
+// Captured variable names for `vector.shuffle` operations follow the L#SH# convention,
+// where L# refers to the level of the tree the shuffle belongs to, and SH# refers to
+// the shuffle index within that level.
+
+func.func @to_from_elements_single_input_shuffle(%a: vector<8xf32>) -> vector<8xf32> {
```

Comment:
Comparing these function names its hard to tell what the difference is:
* `@to_from_elements_single_input_shuffle`,
* `@from_elements_to_elements_single_shuffle`

Wouldn't this be clearer:
* `@single_input`
* `@multiple_inputs` or `@two_inputs`

Ultimately, it's:
```mlir
 // single input
 %1 = vector.from_elements %0#7, %0#0, %0#6, %0#1, %0#5, %0#2, %0#4, %0#3 : vector<8xf32>
```
vs
```mlir
// two inputs
  %2 = vector.from_elements %0#7, %1#0, %0#6, %1#1, %0#5, %1#2, %0#4, %1#3 : vector<8xf32>
```

Reply:
The logic that I've followed is:
  1. Shuffles have two inputs so no need to specify "multi_input" shuffle every time. It's the "default".
  2. Single input shuffle is the exception so it's worth adding the "single_input" tag for it.
  3. Shuffle tree has multiple shuffles in general so no need to specify "multi_suffle". It's the "default"
  4. Single shuffle tree is the exception so it's worth adding the "single_shuffle" tag for it.